### ISSUE TITLE
Camera information about cuhk03

### ISSUE THREAD

**Issue by TJJTJJTJJ** [zhunzhong07/HHL#5]
Hello, I am sorry to bother you again. Your code is awesome and your paper I have read less ten times.

This time, I want to discuss something about cuhk03 dataset. In your paper, 

> Note that images in CUHK03 do not have camera labels, so we cannot perform camera invariance learning. Therefore, we only use CUHK03 as the source domain instead of the target domain. 

And after I consult the CUHK03 dataset, 

> 1467 identities are collected from 5 different pairs of camera views
> 5 x 1 cells and each contains the data collected from a pair of camera views. Each pair of camera views consists of M x 10 cells, where M is the number of identities. For each identity, cell 1-5 are images from one camera and cell 6-10 are images from another camera. However, some identities may have less than 10 images.

So could I think that in CUHK03, there are ten cameras, and each camera could be viewed as its style?

And could I label each image in first pair of camera view as c1 (cell 1-5 )  or c2 (cell 6-10 ), each image in second pair of camera view as c3 (cell 1-5 )   or c4  (cell 6-10 ), and so on?. 

Then use cuhk03 labeled camera information to train stargan, Is that OK? or there is some problems I overlooked? 

Thank you.

**Comment by zhunzhong07** [comment]
Hi, thanks for your interested in our work.

From the description of the CUHK03, we can not sure it is captured from 10 different cameras. For example, the first camera view in cell 1 may be the same with the second camera view in cell3. The best way to verify whether these camera views are different is by comparing the background of them.

Despite this, I think you still could consider them as different cameras and train a StarGan model. Because, if there exists the same camera view, person image will be transferred to the same CamStyle. Or, you could generate the image in other CamStyle if the target camera view is different from the source one.

If you find these cameras are in fact 10 different cameras, please let me know. I will make an explanation.

**Comment by TJJTJJTJJ** [comment]
Hello, I just think about another problem about the training process of stargan during dinner. If I view cuhk03 as ten cameras, but 
> 843*10
> 440*10
> 77*10
> 58*10
> 49*10

in the cell 9 and cell 10, it only has 49*5=245 pictures, that is more less than cell 1 of 843*5=4215 pictures. Is this influence stargan to learn more about style of cell 1 rather than cell 9, 10? That is because I see the origin train code, 

```python
# Train the discriminator
out_src, out_cls = self.D(x_real)
d_loss_real = - torch.mean(out_src)
d_loss_cls = self.classification_loss(out_cls, label_org, self.dataset)
```

during the train discriminator, if images in cell 9 is so less, that make discriminator learn more discrimative skill about cell 1 than cell 9? Maybe I don't make it clear.

The unbalance of numbers of pictures of different cameras will effect training of stargan and IDE? How do you think? It is just my  sudden thought. 

Thank you.

**Comment by zhunzhong07** [comment]
Yeah, it may somewhat influence the results of transferring the images to the style of the cell which has few samples. But I think you could try it.

**Comment by TJJTJJTJJ** [comment]
Thank you, sir. Have a good day.
